tools/devops/automation/scripts/bash/build-nugets.sh

```diff
@@ -18,3 +23,42 @@ DOTNET_PKG_DIR=$(make -C tools/devops print-abspath-variable VARIABLE=DOTNET_PKG
 make -C dotnet package -j
 cp -c "$DOTNET_PKG_DIR"/*.pkg ../package/
 cp -c "$DOTNET_PKG_DIR"/*.msi ../package/
+
+# Down from here we publish mlaunch as .nupkg
+MACCORE_HASH=$(cd "$MACCORE_TOP" && git log -1 --pretty=%h)
+
+if nuget list -source https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet-eng/nuget/v3/index.json -AllVersions -Prerelease Microsoft.DotNet.Mlaunch | grep $MACCORE_HASH; then
+    echo "Mlaunch revision $MACCORE_HASH is already published as nupkg"
+    exit 0
+fi
+
+# Package mlaunch as .nupkg
+echo "Packaging mlaunch revision $MACCORE_HASH as nupkg..."
+
+MLAUNCH_WORK_DIR="$DOTNET_NUPKG_DIR/mlaunch-staging"
+rm -rf "$MLAUNCH_WORK_DIR"
+mkdir -p "$MLAUNCH_WORK_DIR/mlaunch/bin"
+mkdir -p "$MLAUNCH_WORK_DIR/mlaunch/lib/mlaunch"
+
+DOTNET6=$(make -C tools/devops print-abspath-variable VARIABLE=DOTNET6 | grep "^DOTNET6=" | sed -e 's/^DOTNET6=//')
+IOS_DESTDIR=$(make -C tools/devops print-abspath-variable VARIABLE=IOS_DESTDIR | grep "^IOS_DESTDIR=" | sed -e 's/^IOS_DESTDIR=//')
+MONOTOUCH_PREFIX=$(make -C tools/devops print-abspath-variable VARIABLE=MONOTOUCH_PREFIX | grep "^MONOTOUCH_PREFIX=" | sed -e 's/^MONOTOUCH_PREFIX=//')
+
+# Copy mlaunch to staging area
+cp -r "$MACCORE_TOP/tools/mlaunch/Xamarin.Hosting/Xamarin.Launcher/bin/Debug/mlaunch.app" "$MLAUNCH_WORK_DIR/mlaunch/lib/mlaunch"
+cp "$IOS_DESTDIR$MONOTOUCH_PREFIX/bin/mlaunch" "$MLAUNCH_WORK_DIR/mlaunch/bin/"
```

Comment:
Cloning is faster and uses less disk space:

```suggestion
cp -c -r "$MACCORE_TOP/tools/mlaunch/Xamarin.Hosting/Xamarin.Launcher/bin/Debug/mlaunch.app" "$MLAUNCH_WORK_DIR/mlaunch/lib/mlaunch"
cp -c "$IOS_DESTDIR$MONOTOUCH_PREFIX/bin/mlaunch" "$MLAUNCH_WORK_DIR/mlaunch/bin/"
```